# <u>Test Exercice 3 - Model Specification<u>

__<u>Jorge Pineño Pérez<u>__ / Module 4 - Model Specification

This test exercise is of an applied nature and uses data that are available in the data file TestExer3. We consider the so-called Taylor rule for setting the (nominal) interest rate. This model describes the level of the nominal interest rate that the central bank sets as a function of equilibrium real interest rate and inflation, and considers the current level of inflation and production. Taylor (1993) considers the model:

$$i_t = r* + \pi_t + 0.5(\pi_t-\pi*) + 0.5g_t,$$

with it the Federal funds target interest rate at time $t$, $r∗$ the equilibrium real federal funds rate, $π_t$ a measure of inflation, $π∗$ the target inflation rate and $g_t$ the output gap (how much actual output deviates from potential output). We simplify the Taylor rule in two manners. First, we avoid determining $r∗$ and $π∗$ and simply add an intercept to the model to capture these two variables (and any other deviations in the means). Second, we consider production $y_t$ rather than the output gap. In this form the Taylor rule is

$$i_t = \beta_1 + \beta_2 \pi_t + \beta_3 y_t + \epsilon_t.\tag{1}$$

Monthly data are available for the USA over the period 1960 through 2014 for the following variables:

- `INTRATE`: Federal funds interest rate
- `INFL`: Inflation
- `PROD`: Production
- `UNEMPL`: Unemployment
- `COMMPRI`: Commodity prices
- `PCE`: Personal consumption expenditure
- `PERSINC`: Personal income
- `HOUST`: Housing starts

In [55]:
# import packages
install.packages("readxl")
install.packages("dplyr")
install.packages("lmtest")
install.packages("lubridate")
install.packages("tseries")
library(readxl)
library(dplyr)
library(lmtest)
library(lubridate)
library(tseries)

Warning message:
"package 'readxl' is in use and will not be installed"
Warning message:
"package 'dplyr' is in use and will not be installed"
Warning message:
"package 'lmtest' is in use and will not be installed"
Warning message:
"package 'lubridate' is in use and will not be installed"
Installing package into 'C:/Users/USER/AppData/Local/R/win-library/4.4'
(as 'lib' is unspecified)

also installing the dependencies 'xts', 'TTR', 'quadprog', 'quantmod'




package 'xts' successfully unpacked and MD5 sums checked
package 'TTR' successfully unpacked and MD5 sums checked
package 'quadprog' successfully unpacked and MD5 sums checked
package 'quantmod' successfully unpacked and MD5 sums checked
package 'tseries' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\USER\AppData\Local\Temp\RtmpuuFHD3\downloaded_packages


Registered S3 method overwritten by 'quantmod':
  method            from
  as.zoo.data.frame zoo 



In [40]:
# import the stock index dataframe
fred <- read_excel('TestExer-3-TaylorRule-round1.xlsx')
head(fred)

OBS    INTRATE INFL    PROD     UNEMPL  COMMPRI   PCE     PERSINC HOUST    
1 1960:1 3.99    1.24095 10.03653 3.41845   7.95262 5.70962 1.68419 -11.88896
2 1960:2 3.97    1.41379  6.96248 3.46575  -8.55856 5.06452 1.33094  -9.83803
3 1960:3 3.84    1.51881  4.49681 2.71993 -16.83599 5.55733 0.89195 -31.54321
4 1960:4 3.92    1.93237  1.50624 2.79820  -5.03145 7.77351 0.67636 -18.93082
5 1960:5 3.85    1.82507 -0.11398 1.72552 -12.44240 4.39179 0.33667 -15.15354
6 1960:6 3.32    1.71762 -1.48262 1.24443 -24.31655 3.73509 0.06719 -17.03260

__(a)__ Use general-to-specific to come to a model. Start by regressing the federal funds rate on the other 7 variables and eliminate 1 variable at a time.

In [3]:
# start by regressing "INTRATE" on the other 7 variables
intrate1 <- lm(INTRATE ~ INFL + PROD + UNEMPL + COMMPRI + PCE + PERSINC + HOUST, data=fred)
summary(intrate1)


Call:
lm(formula = INTRATE ~ INFL + PROD + UNEMPL + COMMPRI + PCE + 
    PERSINC + HOUST, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.4066 -1.4340 -0.1175  1.3555  7.7386 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.221161   0.244995  -0.903   0.3670    
INFL         0.696059   0.062229  11.185  < 2e-16 ***
PROD        -0.057743   0.039900  -1.447   0.1483    
UNEMPL       0.102481   0.096757   1.059   0.2899    
COMMPRI     -0.005521   0.002974  -1.857   0.0638 .  
PCE          0.344380   0.069455   4.958 9.08e-07 ***
PERSINC      0.246999   0.060590   4.077 5.13e-05 ***
HOUST       -0.019411   0.004672  -4.155 3.68e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.188 on 652 degrees of freedom
Multiple R-squared:  0.6385,	Adjusted R-squared:  0.6346 
F-statistic: 164.5 on 7 and 652 DF,  p-value: < 2.2e-16


In [4]:
# eliminate the variable with the highest P-value and repeat the process
intrate1 <- lm(INTRATE ~ INFL + PROD + COMMPRI + PCE + PERSINC + HOUST, data=fred) # eliminate "UNEMPL"
summary(intrate1)


Call:
lm(formula = INTRATE ~ INFL + PROD + COMMPRI + PCE + PERSINC + 
    HOUST, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.5322 -1.4982 -0.1005  1.3882  7.6954 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.290851   0.236016  -1.232   0.2183    
INFL         0.693309   0.062180  11.150  < 2e-16 ***
PROD        -0.025460   0.025752  -0.989   0.3232    
COMMPRI     -0.006514   0.002822  -2.308   0.0213 *  
PCE          0.368561   0.065602   5.618 2.86e-08 ***
PERSINC      0.251581   0.060441   4.162 3.57e-05 ***
HOUST       -0.021023   0.004417  -4.760 2.39e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.188 on 653 degrees of freedom
Multiple R-squared:  0.6379,	Adjusted R-squared:  0.6346 
F-statistic: 191.7 on 6 and 653 DF,  p-value: < 2.2e-16


In [5]:
# eliminate the variable with the highest P-value and repeat the process
intrate1 <- lm(INTRATE ~ INFL + COMMPRI + PCE + PERSINC + HOUST, data=fred) # eliminate "PROD"
summary(intrate1)


Call:
lm(formula = INTRATE ~ INFL + COMMPRI + PCE + PERSINC + HOUST, 
    data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.1631 -1.5244 -0.1125  1.3715  7.6725 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.240119   0.230366  -1.042  0.29764    
INFL         0.717527   0.057152  12.555  < 2e-16 ***
COMMPRI     -0.007501   0.002640  -2.841  0.00464 ** 
PCE          0.340525   0.059156   5.756 1.32e-08 ***
PERSINC      0.240242   0.059342   4.048 5.77e-05 ***
HOUST       -0.020530   0.004389  -4.678 3.52e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.188 on 654 degrees of freedom
Multiple R-squared:  0.6374,	Adjusted R-squared:  0.6346 
F-statistic: 229.9 on 5 and 654 DF,  p-value: < 2.2e-16


The final model we get using the general-to-specific approach is

$$ i_t = -0.24 + 0.72\pi_t - 0.01P_t + 0.34C_t + 0.24Y_t - 0.02H_t $$

Where

$\pi_t =$ Inflation at year $t$

$P_t =$ Commodity prices at year $t$

$C_t =$ Personal consumption expenditure at year $t$

$Y_t =$ Personal income at year $t$

$H_t =$ Housing starts at year $t$

__(b)__ Use specific-to-general to come to a model. Start by regressing the federal funds rate on only a constant and add 1 variable at a time. Is the model the same as in (a)?

In [8]:
# start by regressing "INTRATE" on only a constant
intrate2 <- lm(INTRATE ~ 1, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ 1, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.2776 -2.3476 -0.1276  1.7749 13.7524 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   5.3476     0.1409   37.96   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.619 on 659 degrees of freedom


In [12]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.9947 -1.5592  0.0603  1.4990  8.2910 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  1.64209    0.15863   10.35   <2e-16 ***
INFL         0.94534    0.03268   28.93   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.403 on 658 degrees of freedom
Multiple R-squared:  0.5598,	Adjusted R-squared:  0.5591 
F-statistic: 836.6 on 1 and 658 DF,  p-value: < 2.2e-16


In [13]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + PROD, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + PROD, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.1592 -1.6762  0.0141  1.3730  7.9203 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  1.24890    0.17619   7.088 3.51e-12 ***
INFL         0.97498    0.03273  29.785  < 2e-16 ***
PROD         0.09472    0.01971   4.805 1.92e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.364 on 657 degrees of freedom
Multiple R-squared:  0.5747,	Adjusted R-squared:  0.5734 
F-statistic: 443.9 on 2 and 657 DF,  p-value: < 2.2e-16


In [14]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + PROD + UNEMPL, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + PROD + UNEMPL, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.1064 -1.6080 -0.0871  1.2663  8.8865 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  1.20414    0.17217   6.994 6.61e-12 ***
INFL         0.89127    0.03508  25.404  < 2e-16 ***
PROD        -0.07989    0.03582  -2.230   0.0261 *  
UNEMPL       0.49783    0.08613   5.780 1.15e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.308 on 656 degrees of freedom
Multiple R-squared:  0.5953,	Adjusted R-squared:  0.5935 
F-statistic: 321.7 on 3 and 656 DF,  p-value: < 2.2e-16


In [15]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + PROD + UNEMPL + COMMPRI, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + PROD + UNEMPL + COMMPRI, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.2604 -1.5909 -0.0619  1.3232  8.7780 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  1.194349   0.171870   6.949 8.89e-12 ***
INFL         0.900884   0.035347  25.487  < 2e-16 ***
PROD        -0.041463   0.040741  -1.018   0.3092    
UNEMPL       0.434899   0.091716   4.742 2.60e-06 ***
COMMPRI     -0.006087   0.003098  -1.965   0.0498 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.303 on 655 degrees of freedom
Multiple R-squared:  0.5977,	Adjusted R-squared:  0.5952 
F-statistic: 243.3 on 4 and 655 DF,  p-value: < 2.2e-16


In [16]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + UNEMPL + COMMPRI + PCE, data=fred) # eliminate "PROD" (p-value = 0.3092) 
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + UNEMPL + COMMPRI + PCE, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.7901 -1.6664 -0.0672  1.5273  7.9057 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.187215   0.239572   0.781  0.43482    
INFL         0.729720   0.045579  16.010  < 2e-16 ***
UNEMPL       0.139100   0.059730   2.329  0.02017 *  
COMMPRI     -0.009967   0.002691  -3.704  0.00023 ***
PCE          0.306432   0.054839   5.588 3.37e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.251 on 655 degrees of freedom
Multiple R-squared:  0.6154,	Adjusted R-squared:  0.613 
F-statistic:   262 on 4 and 655 DF,  p-value: < 2.2e-16


In [17]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + UNEMPL + COMMPRI + PCE + PERSINC, data=fred)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + UNEMPL + COMMPRI + PCE + PERSINC, 
    data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.6924 -1.4698 -0.1248  1.4841  7.8090 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.063520   0.238067   0.267  0.78969    
INFL         0.853378   0.053203  16.040  < 2e-16 ***
UNEMPL       0.078693   0.060544   1.300  0.19414    
COMMPRI     -0.008600   0.002673  -3.217  0.00136 ** 
PCE          0.185479   0.060834   3.049  0.00239 ** 
PERSINC      0.265829   0.061123   4.349 1.59e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.221 on 654 degrees of freedom
Multiple R-squared:  0.6262,	Adjusted R-squared:  0.6233 
F-statistic: 219.1 on 5 and 654 DF,  p-value: < 2.2e-16


In [18]:
# add the next variable in the dataframe and repeat the process
intrate2 <- lm(INTRATE ~ INFL + COMMPRI + PCE + PERSINC + HOUST, data=fred) # eliminate "UNEMPL" (p-value = 0.19414)
summary(intrate2)


Call:
lm(formula = INTRATE ~ INFL + COMMPRI + PCE + PERSINC + HOUST, 
    data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.1631 -1.5244 -0.1125  1.3715  7.6725 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.240119   0.230366  -1.042  0.29764    
INFL         0.717527   0.057152  12.555  < 2e-16 ***
COMMPRI     -0.007501   0.002640  -2.841  0.00464 ** 
PCE          0.340525   0.059156   5.756 1.32e-08 ***
PERSINC      0.240242   0.059342   4.048 5.77e-05 ***
HOUST       -0.020530   0.004389  -4.678 3.52e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.188 on 654 degrees of freedom
Multiple R-squared:  0.6374,	Adjusted R-squared:  0.6346 
F-statistic: 229.9 on 5 and 654 DF,  p-value: < 2.2e-16


The final model we get using the general-to-specific approach is

$$ i_t = -0.24 + 0.72\pi_t - 0.01P_t + 0.34C_t + 0.24Y_t - 0.02H_t $$

As we can see, this model is the same as the model generated with the general-to-specific approach used in (a).

__(c)__ Compare your model from (a) and the Taylor rule of equation (1). Consider $R^2$, AIC and BIC. Which of the models do you prefer?

In [20]:
# regress "INFRATE" on a constant, "INFL", and "PROD", like in equation (1)
taylor <- lm(INTRATE ~ INFL + PROD, data=fred)
summary(taylor)


Call:
lm(formula = INTRATE ~ INFL + PROD, data = fred)

Residuals:
    Min      1Q  Median      3Q     Max 
-5.1592 -1.6762  0.0141  1.3730  7.9203 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  1.24890    0.17619   7.088 3.51e-12 ***
INFL         0.97498    0.03273  29.785  < 2e-16 ***
PROD         0.09472    0.01971   4.805 1.92e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 2.364 on 657 degrees of freedom
Multiple R-squared:  0.5747,	Adjusted R-squared:  0.5734 
F-statistic: 443.9 on 2 and 657 DF,  p-value: < 2.2e-16


The model we get using the Taylor rule is

$$ i_t = 1.25 + 0.97\pi_t + 0.09Y_t $$

Where

$\pi_t =$ Inflation at year $t$

$Y_t =$ Production at year $t$

In [21]:
# get R squared, AIC, and BIC of the model in (a)
intrate1_r2 <- summary(intrate1)$r.squared
intrate1_aic <- AIC(intrate1)
intrate1_bic <- BIC(intrate1)

# get R squared, AIC, and BIC of the Taylor rule model
taylor_r2 <- summary(taylor)$r.squared
taylor_aic <- AIC(taylor)
taylor_bic <- BIC(taylor)

In [23]:
# display the results for the model in (a)
results_table1 <- data.frame(
  R2 = intrate1_r2,
  AIC = intrate1_aic,
  BIC = intrate1_bic
)
print(results_table1)

        R2      AIC      BIC
1 0.637361 2914.423 2945.869


In [24]:
# display the results for the Taylor rule model
results_table2 <- data.frame(
  R2 = taylor_r2,
  AIC = taylor_aic,
  BIC = taylor_bic
)
print(results_table2)

         R2      AIC      BIC
1 0.5747014 3013.616 3031.585


As we can see, and based on the $R^2$, AIC, and BIC of the two regressions, the model from (a) would be preferable in this case, as it has a higher $R^2$, and lower AIC and BIC.

__(d)__ Test the Taylor rule of equation (1) using the RESET test, Chow break and forecast test (with in both tests as break date January 1980) and a Jarque-Bera test. What do you conclude?

In [50]:
# perform the RESET test on the Taylor rule regression of (c)
taylor_reset <- resettest(taylor)
taylor_reset


	RESET test

data:  taylor
RESET = 2.2578, df1 = 2, df2 = 655, p-value = 0.1054


F-test for the Chow Break Statistic:

$$F = \frac{(SSR_{\text{pooled}} - (SSR_1 + SSR_2)) \ \big/ \ k}{(SSR_1 + SSR_2) \ \big/ \ (n_1 + n_2 - 2k)}$$

In [41]:
# convert "OBS" column of "fred" to <date> format
fred <- fred %>%
  mutate(OBS=ym(OBS))
head(fred)

OBS        INTRATE INFL    PROD     UNEMPL  COMMPRI   PCE     PERSINC
1 1960-01-01 3.99    1.24095 10.03653 3.41845   7.95262 5.70962 1.68419
2 1960-02-01 3.97    1.41379  6.96248 3.46575  -8.55856 5.06452 1.33094
3 1960-03-01 3.84    1.51881  4.49681 2.71993 -16.83599 5.55733 0.89195
4 1960-04-01 3.92    1.93237  1.50624 2.79820  -5.03145 7.77351 0.67636
5 1960-05-01 3.85    1.82507 -0.11398 1.72552 -12.44240 4.39179 0.33667
6 1960-06-01 3.32    1.71762 -1.48262 1.24443 -24.31655 3.73509 0.06719
  HOUST    
1 -11.88896
2  -9.83803
3 -31.54321
4 -18.93082
5 -15.15354
6 -17.03260

In [44]:
# store the sum of square residuals of "taylor"
ssr_taylor <- sum((resid(taylor))^2)

# run a regression from 1960 to 1979 and store the sum of square residuals
taylor1 <- lm(INTRATE ~ INFL + PROD, data=(fred %>% filter(between(OBS, as.Date('1960-01-01'), as.Date('1979-12-01')))))
ssr_taylor1 <- sum((resid(taylor1))^2)

# run a regression from 1980 to 2014 and store the sum of square residuals
taylor2 <- lm(INTRATE ~ INFL + PROD, data=(fred %>% filter(between(OBS, as.Date('1980-01-01'), as.Date('2014-12-01')))))
ssr_taylor2 <- sum((resid(taylor2))^2)

# get the rest of variables we need for computing the Chow Break Test
k = length(coef(taylor))
n1 = nrow(fred %>% filter(between(OBS, as.Date('1960-01-01'), as.Date('1979-12-01'))))
n2 = nrow(fred %>% filter(between(OBS, as.Date('1980-01-01'), as.Date('2014-12-01'))))

In [48]:
# perform the Chow break test following the formula above
taylor_cb = ((ssr_taylor-(ssr_taylor1+ssr_taylor2))/k)/((ssr_taylor1+ssr_taylor2)/(n1+n2-2*k))
taylor_cb

[1] 28.73501

F-test for the Chow forecast test:

$$F = \frac{(SSR_{\text{pooled}} - SSR_1) \ \big/ \ n_2}{SSR_1 \ \big/ \ (n_1 - k)}$$

In [51]:
# perform the Chow forecast test following the formula above
taylor_cf = ((ssr_taylor-ssr_taylor1)/n2)/(ssr_taylor1/(n1-k))
taylor_cf

[1] 5.510518

In [57]:
# perform the Jarque-Bera test
taylor_jb <- jarque.bera.test(resid(taylor))
taylor_jb


	Jarque Bera Test

data:  resid(taylor)
X-squared = 12.444, df = 2, p-value = 0.001985


In [66]:
# display the results
results_table3 <- data.frame(
  'Chow_break_test' = taylor_cb,
  'Chow_forecast_test' = taylor_cf,
  'Jarque-Bera_test' = taylor_jb$p.value
)
print(results_table3)

  Chow_break_test Chow_forecast_test Jarque.Bera_test
1        28.73501           5.510518      0.001985228


Looking at the values from the Chow break and forecast tests, and the Jarque-Bera test, we can draw some conclusions from the model using the Taylor rule of equation (1):

1. There is strong evidence of a break around 1980, given the high value of the Chow break test, well above the critical value for a 5% significance level

2. Although lower than the break test, the Chow forecast test's value of 5.51 also points to the fact that the predicitve capabilities of the model are low for observations from 1980 onwards

3. The model's residuals are not normally distributed, given the low value of the Jarque-Bera test

From these three tests, we can conclude that there was likely a change in monetary policy between the pre-1980 and post-1980 periods.